File: frontend/src/types/customer.ts
export type CustomerType = 'INDIVIDUAL' | 'FAMILY' | 'OFFICE' | 'OTHER'
export type DietType = 'VEG' | 'NON_VEG' | 'EGG' | 'CUSTOM'
export type MealSession = 'BREAKFAST' | 'LUNCH' | 'DINNER'
export type SpiceLevel = 'MILD' | 'NORMAL' | 'SPICY'

export interface Address {
  address_line: string
  area: string
  landmark?: string
  city: string
  pincode?: string
  latitude?: number
  longitude?: number
  google_place_id?: string
  formatted_address?: string
}

export interface MealPreference {
  diet_type: DietType
  sessions: MealSession[]
  spice_level: SpiceLevel
  rice_preference?: string
  food_exclusions?: string
  allergy_note?: string
  delivery_instructions?: string
}

export interface Customer {
  id: string
  name: string
  phone: string
  alternate_phone?: string
  whatsapp_number?: string
  customer_type: CustomerType
  is_active: boolean
  address: Address
  meal_preference: MealPreference
  notes?: string
  created_at: string
  updated_at?: string
}

export interface CustomerCreateInput {
  name: string
  phone: string
  alternate_phone?: string
  whatsapp_number?: string
  customer_type: CustomerType
  is_active?: boolean
  address: Address
  meal_preference: MealPreference
  notes?: string
}

export type EnquiryStatus =
  | 'NEW'
  | 'CONTACTED'
  | 'QUOTED'
  | 'FOLLOW_UP'
  | 'CONFIRMED'
  | 'REJECTED'
  | 'CONVERTED'
  | 'CLOSED'

export interface Enquiry {
  id: string
  customer_name: string
  phone: string
  requested_meal?: string
  requested_date?: string
  number_of_persons: number
  diet_type: DietType
  subscription_type: string
  delivery_location: string
  quoted_price?: number
  notes?: string
  follow_up_date?: string
  assigned_admin?: string
  status: EnquiryStatus
  customer_id?: string
  enquiry_date: string
  created_at: string
}

export interface EnquiryCreateInput {
  customer_name: string
  phone: string
  requested_meal?: string
  requested_date?: string
  number_of_persons: number
  diet_type: DietType
  subscription_type?: string
  delivery_location: string
  quoted_price?: number
  notes?: string
  follow_up_date?: string
  assigned_admin?: string
  status?: EnquiryStatus
}

export interface MealPlan {
  id: string
  name: string
  diet_type: DietType
  meal_sessions: MealSession[]
  price: number
  billing_cycle: string
  service_days: number
  delivery_eligible: boolean
  is_active: boolean
  description?: string
  created_at: string
}

export interface MealPlanCreateInput {
  name: string
  diet_type: DietType
  meal_sessions: MealSession[]
  price: number
  billing_cycle?: string
  service_days?: number
  delivery_eligible?: boolean
  is_active?: boolean
  description?: string
}

File: backend/app/schemas/customer.py
from enum import Enum
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict


class CustomerType(str, Enum):
    INDIVIDUAL = "INDIVIDUAL"
    FAMILY = "FAMILY"
    OFFICE = "OFFICE"
    OTHER = "OTHER"


class DietType(str, Enum):
    VEG = "VEG"
    NON_VEG = "NON_VEG"
    EGG = "EGG"
    CUSTOM = "CUSTOM"


class MealSession(str, Enum):
    BREAKFAST = "BREAKFAST"
    LUNCH = "LUNCH"
    DINNER = "DINNER"


class SpiceLevel(str, Enum):
    MILD = "MILD"
    NORMAL = "NORMAL"
    SPICY = "SPICY"


class AddressSchema(BaseModel):
    address_line: str = Field(..., description="Street address / door number")
    area: str = Field(..., description="Area / locality (e.g. Anna Nagar)")
    landmark: Optional[str] = None
    city: str = Field(default="Chennai")
    pincode: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    google_place_id: Optional[str] = None
    formatted_address: Optional[str] = None


class MealPreferenceSchema(BaseModel):
    diet_type: DietType = DietType.VEG
    sessions: List[MealSession] = Field(default_factory=lambda: [MealSession.LUNCH])
    spice_level: SpiceLevel = SpiceLevel.NORMAL
    rice_preference: Optional[str] = None
    food_exclusions: Optional[str] = None
    allergy_note: Optional[str] = None
    delivery_instructions: Optional[str] = None


class CustomerBase(BaseModel):
    name: str = Field(..., min_length=2, max_length=100)
    phone: str = Field(..., min_length=8, max_length=20)
    alternate_phone: Optional[str] = None
    whatsapp_number: Optional[str] = None
    customer_type: CustomerType = CustomerType.INDIVIDUAL
    is_active: bool = True
    address: AddressSchema
    meal_preference: MealPreferenceSchema = Field(default_factory=MealPreferenceSchema)
    notes: Optional[str] = None


class CustomerCreate(CustomerBase):
    pass


class CustomerUpdate(BaseModel):
    name: Optional[str] = None
    phone: Optional[str] = None
    alternate_phone: Optional[str] = None
    whatsapp_number: Optional[str] = None
    customer_type: Optional[CustomerType] = None
    is_active: Optional[bool] = None
    address: Optional[AddressSchema] = None
    meal_preference: Optional[MealPreferenceSchema] = None
    notes: Optional[str] = None


class CustomerStatusUpdate(BaseModel):
    is_active: bool


class CustomerResponse(CustomerBase):
    id: str
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)
